### ISSUE TITLE
advice on multiple calls to couchbase

### ISSUE THREAD

**Issue by chenbekor** [ReactiveCouchbase/ReactiveCouchbase-play#23]
hi Mathieu,

I wanted to consult with you on how to best use reactivecouchbase to do the following:

My use case is (using example entities for the sake of discussion):
1. Fetch User
2. From user get Account ID  and fetch Account
3. Fetch Beers owned by Account
4. Combine all above and return as json to calling client

Use case might be more evolved but I keep it simple for the purpose of this discussion.

In order to manage to do the above I need to perform a series of back to back typed fetching (non json) from couch as follows:

For example bucket.get[User](id)  which returns with a monad of type Future[Option[User]]

As you know - I need to now "extract" the User via  "map" for handling the future and than  the Option with another map and a getOrElse.

So long story short - I need to use a series of map, flatMap and getOrElse all intermingled in inner calls. The code looks like horror movie. Is that the price one has to pay for going non blocking? ;)

Maybe you can show as an example how would you implement such complex workflows while keeping the Scala code concise elegant and easy to read / understand / enhance...

thanks agains Mathieu for the continuos support !!
